### PR TITLE
feat(query): Support create network policy

### LINKED ISSUE #11872: Feature: Support network policies for the user to control access IP address
**Summary**

Support [network policies](https://docs.snowflake.com/en/user-guide/network-policies) for the user to control access IP address

### PR DESCRIPTION
**Pull request by b41sh**
I hereby agree to the terms of the CLA available at: https://databend.rs/dev/policies/cla/

## Summary

Support Create/Alter/Drop/Desc/Show network policy

- Closes #11872

### CODE REVIEW

**Comment by drmingdrmer** [comment]
### PR summary(by Claude):

 We can summarize the main points as:

* **Updated CIDR dependency**  
  Added the cidr crate which enables parsing of IPv4 CIDR ranges. Used in validating the ALLOWED_IP_LIST and BLOCKED_IP_LIST while binding network policy statements. Adds around 20 lines of code.

* **Added network policy DDL statements**    
  Added AST nodes, parsers, binders and planners for `CREATE NETWORK POLICY`, `ALTER NETWORK POLICY`, `DROP NETWORK POLICY` and `DESCRIBE NETWORK POLICY`. Code changes amount to around 400 lines.

* **Added management APIs for network policies**  
  Added APIs to add, update, drop and get network policies. Used by the interpreters to execute the DDL statements. Adds around 170 lines of code.

* **Implemented network policy interpreters**   
  Added interpreters to execute the network policy DDL statements. Each interpreter calls the corresponding management API. Adds around 250 lines of code.

* **Added SHOW NETWORK POLICIES statement**   
  Added an AST node, parser, binder and planner for `SHOW NETWORK POLICIES`. The interpreter fetches all network policies from the management API and returns the result. Adds around 100 lines of code.

How does this summary look? Let me know if you want me to modify or expand any part.

File: src/meta/app/src/principal/network_policy.rs [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+    fn try_from(value: Vec<u8>) -> Result<Self, Self::Error> {
+        match serde_json::from_slice(&value) {
+            Ok(network_policy) => Ok(network_policy),
+            Err(serialize_error) => Err(NetworkPolicySerdeError {
+                message: "Cannot deserialize NetworkPolicy from bytes".to_string(),
+                source: AnyError::new(&serialize_error),
+            }),
+        }
+    }
+}
```
Comment (drmingdrmer) [inline src/meta/app/src/principal/network_policy.rs:52]: Why does NetworkPolicy needs to be able to decode from json?

**Review by drmingdrmer — COMMENTED** [review]
Where is NetworkPolicy used? Does it belong to another meta-data type or is there an meta-service API using it?

**Comment by b41sh** [comment]
> Where is NetworkPolicy used? Does it belong to another meta-data type or is there an meta-service API using it?

Network policy is related to the user and is used to control the ip from which the user is allowed to log in. Set network policy for users will be implemented in the next PR.
for example
```sql
ALTER USER test SET NETWORK_POLICY = mypolicy;
ALTER USER test UNSET NETWORK_POLICY;
```

File: src/meta/app/src/principal/network_policy.rs (hunk 1 above)
Comment (b41sh) [reply src/meta/app/src/principal/network_policy.rs:52]: This function is not needed anymore, I will remove this code.

**Review by drmingdrmer — APPROVED** [review]

**Review by flaneur2020 — APPROVED** [review]
LGTM

**Review by sundy-li — APPROVED** [review]

**Pull request merged**